drivers/src/ns16550.rs

```diff
@@ -1,17 +1,22 @@
 //! Driver for the NS16550 UART chip.
 //! The datasheet used to write this is: <http://caro.su/msx/ocm_de1/16550.pdf>
 
+use super::lock::Lock;
 use super::Console;
 use super::Driver;
 
 const TRANSMITTER_HOLDING_REGISTER: usize = 0;
 const _INTERRUPT_ENABLE_REGISTER: usize = 1;
 
 pub struct Ns16550 {
+    inner: Lock<Ns16550Inner>,
+}
+
+struct Ns16550Inner {
     base_register_address: usize,
```

Comment:
It feels a bit weird to have the base address which will be a constant behind a lock

Reply:
Yes ? 
I see it more as, all the operations than can be done on the device are behind a lock.
We don't really ever need to just access the base right ?
I guess most functions in the inner could take a base parameter then ?

Reply:
> We don't really ever need to just access the base right ?

`get_address_range` just need the base

> I guess most functions in the inner could take a base parameter then ?

Yeah that would be even weirder... Let keep it as is then

Reply:
I agree about that method, when we'll have a RwLock it'll be ok